lib/presentation/forget_password/forget_password_view.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:online_exam_app/core/base/base_view.dart';
+import 'package:online_exam_app/core/constants/routes.dart';
+import 'package:online_exam_app/core/di/di.dart';
+import 'package:online_exam_app/core/utils/app_dialogs.dart';
+import 'package:online_exam_app/presentation/forget_password/Widgets/forget_password_form.dart';
+import 'package:online_exam_app/presentation/forget_password/forget_password_contract.dart';
+import 'package:online_exam_app/presentation/forget_password/forget_password_view_model.dart';
+
+class ForgetPasswordView extends StatefulWidget {
+  const ForgetPasswordView({super.key});
+
+  @override
+  State<ForgetPasswordView> createState() => _ForgetPasswordViewState();
+}
+
+class _ForgetPasswordViewState
+    extends BaseState<ForgetPasswordView, ForgetPasswordViewModel> {
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return BlocProvider(
+      create: (context) => viewModel,
+      child: BlocConsumer<ForgetPasswordViewModel, ForgetPasswordViewState>(
+        listener: (context, state) {
+          if (state is ForgetPasswordLoadingState) {
+            AppDialogs.showLoading(
+                message: viewModel.locale!.loading, context: context);
+          }
+          if (state is ForgetPasswordSuccessState) {
+            AppDialogs.showSuccessDialog(
+                message: viewModel.locale!.otpSentToYourEmail,
+                context: context,
+                posActionTitle: viewModel.locale!.ok,
+                posAction: () {
+                  Navigator.pushReplacementNamed(context, Routes.otpVerifyRoute,
+                    arguments: viewModel.emailController.text
+                  );
+                });
+          }
```

Comment:
this is mvvm violation because this code make view apply logic you should make action then call this action for navigation and pass the arguments in the state emitted